File: Spy_Detected.py
t = int(input())
ans = 0
while(t>0):
    t = t - 1
    n = int(input())
    a = dict()
    ai = 1
    for i in range(1,n+1):
        bi = int(input())
        a[ai] = bi
        ai+=1
    count = dict()
    for i in a:
        if a[i]  not in count:
            count[a[i]] = 1
        else:
            count[a[i]]+=1
    max = -1
    index = 0
    for i in count:
        if(max<count[i]):
            max = count[i]
            index = i
    for i in a:
        if(a[i] != index):
            ans = i
            break
else:
    print(ans)

File: SpyDetected.cpp
#include <iostream>
using namespace std;
int main()
{
    int t;
    cin>>t;
    while(t>0)
    {
        t--;
        int n,index;
        cin>>n;
        int array[n];
        for(int i=0;i<n;i++)
        {
            cin>>array[i];
        }
        for(int i=0;i<n;i++)
        {
            if(i == 0)
            {
                if(array[i] != array[i+1] && array[i+1] == array[i+2])
                {
                    index = i;
                }
            }
            else if(i == (n-1))
            {
                if(array[i] != array[i-1] && array[i-1] == array[i-2])
                {
                    index = i;
                }
            }
            else{
                if((array[i-1] != array[i]) && (array[i] != array[i+1]) && (array[i-1] == array[i+1]))
                {
                    index = i;
                }
            }
        }
        cout<<(index+1)<<endl;
    }
    return 0;
}               